protocol/consistencychecks.go

```diff
@@ -199,8 +195,8 @@ func (cc *ConsistencyChecks) verifyRegistration(msg *Response,
 func (cc *ConsistencyChecks) verifyKeyLookup(msg *Response,
 	uname string, key []byte) error {
 	df := msg.DirectoryResponse.(*DirectoryProof)
-	ap := df.AP
-	str := df.STR
+	ap := df.AP[0]
+	str := df.STR[0]
```

Comment:
I think we need to validate `df`at this point and fail early if `len(df.AP) != 1 || len(df.STR) != 1`. For now, we should do the same in `verifyRegistration` although this will probably change when we verify the prior history on a registration.